jar-infer/src/main/java/com/uber/nullaway/jarinfer/DefinitelyDerefedParamsDriver.java

```diff
@@ -161,6 +175,58 @@ private static String extractJAR(String jarPath) {
     return jarDir;
   }
 
+  /*
+   * Repack JAR archive and return path
+   *
+   */
+  private static String packJAR(String jarDir) {
+    Preconditions.checkArgument(
+        Files.isDirectory(Paths.get(jarDir)), "invalid jar directory!" + jarDir);
+    String jarPath = jarDir + ".ji.jar";
+    System.out.println("repacking " + jarPath + "...");
+    File jarDirFile = new File(jarDir);
+    try {
+      FileOutputStream fos = new FileOutputStream(jarPath);
+      JarOutputStream jos = new JarOutputStream(fos, new Manifest());
+      final IOFileFilter jarFilter =
+          new IOFileFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+              return !(name.endsWith(".DS_Store") || name.endsWith("MANIFEST.MF"));
```

Comment:
Curious why we are excluding `MANIFEST.MF`, does it get created by `JarOutputStream`? (I see `new Manifest()` up there, so I assume that's what's happening, but just curious)

Reply:
JarOutputStream creates the MANIFEST.MF file.